src/ovs/aca_ovs_programmer.cpp

```diff
@@ -34,51 +39,78 @@ ACA_OVS_Programmer &ACA_OVS_Programmer::get_instance()
 
-  // TODO: see if br-int and br-tun is already there, one option is the delete
-  // them and start everything from scratch
+    execute_ovsdb_command("add-br br-tun", not_care_culminative_time, overall_rc);
 
-  // create br-int and br-tun bridges
-  execute_ovsdb_command("add-br br-int", not_care_culminative_time, overall_rc);
+    // execute_openflow_command("del-flows br-tun", not_care_culminative_time, overall_rc);
 
-  execute_ovsdb_command("add-br br-tun", not_care_culminative_time, overall_rc);
+    // create and connect the patch ports between br-int and br-tun
+    execute_ovsdb_command("add-port br-int patch-tun", not_care_culminative_time, overall_rc);
 
-  execute_openflow_command("del-flows br-tun", not_care_culminative_time, overall_rc);
+    execute_ovsdb_command("set interface patch-tun type=patch",
+                          not_care_culminative_time, overall_rc);
 
-  // create and connect the patch ports between br-int and br-tun
-  execute_ovsdb_command("add-port br-int patch-tun", not_care_culminative_time, overall_rc);
+    execute_ovsdb_command("set interface patch-tun options:peer=patch-int",
+                          not_care_culminative_time, overall_rc);
 
-  execute_ovsdb_command("set interface patch-tun type=patch",
-                        not_care_culminative_time, overall_rc);
+    execute_ovsdb_command("add-port br-tun patch-int", not_care_culminative_time, overall_rc);
 
-  execute_ovsdb_command("set interface patch-tun options:peer=patch-int",
-                        not_care_culminative_time, overall_rc);
+    execute_ovsdb_command("set interface patch-int type=patch",
+                          not_care_culminative_time, overall_rc);
 
-  execute_ovsdb_command("add-port br-tun patch-int", not_care_culminative_time, overall_rc);
+    execute_ovsdb_command("set interface patch-int options:peer=patch-tun",
+                          not_care_culminative_time, overall_rc);
 
-  execute_ovsdb_command("set interface patch-int type=patch",
-                        not_care_culminative_time, overall_rc);
+    // adding default flows
+    execute_openflow_command("add-flow br-tun \"table=0, priority=1,in_port=\"patch-int\" actions=resubmit(,2)\"",
+                             not_care_culminative_time, overall_rc);
 
-  execute_ovsdb_command("set interface patch-int options:peer=patch-tun",
-                        not_care_culminative_time, overall_rc);
+    execute_openflow_command("add-flow br-tun \"table=2, priority=0 actions=resubmit(,22)\"",
+                             not_care_culminative_time, overall_rc);
 
-  // adding default flows
-  // table 2 is for gre, should send to table 3
-  execute_openflow_command("add-flow br-tun \"table=0, priority=1,in_port=\"patch-int\" actions=resubmit(,2)\"",
-                           not_care_culminative_time, overall_rc);
+  } else {
+    // case 3: only one of the br-int or br-tun is there,
```

Comment:
Can we go ahead to create the missing br here?

Reply:
I thought about that option but not sure if it is a good idea. Because there are patch ports and patch connections between br-int and br-tun. In the weird configuration when only one of the br-int and br-tun is there, do we recreate the patch connection and ports on both sides? I am concerned that if we limp along for this problematic environment, things may not work down correctly the line and it is too hard to trace back to this environment setup issue.

Reply:
I see if this scenario need additional attention as it is too troublesome, one possibility is to log a more severe error like critical so that it could easier to catch people's attention during livesite investigation.

Reply:
yes, I have changed the logging to ACA_LOG_CRIT